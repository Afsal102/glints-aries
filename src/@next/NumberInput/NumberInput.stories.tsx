import React from 'react';
import { Story, Meta } from '@storybook/react';
import { BaseContainer } from '../../Layout/GlintsContainer/GlintsContainer';
import { NumberInputProps, NumberInput } from './NumberInput';

(NumberInput as React.FunctionComponent<NumberInputProps>).displayName =
  'Number';

export default {
  title: '@next/NumberInput',
  component: NumberInput,
  decorators: [Story => <BaseContainer>{Story()}</BaseContainer>],
  argTypes: {},
} as Meta;

const Template: Story<NumberInputProps> = args => <NumberInput {...args} />;

export const Interactive = Template.bind({});
Interactive.args = {
  placeholder: '0.0',
  disabled: false,
};
