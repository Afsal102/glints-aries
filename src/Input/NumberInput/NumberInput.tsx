import * as React from 'react';
import TextField, { TextFieldProps } from '../TextField';

const NumberInput = (props: Props) => {
  return <TextField {...props} type="number" />;
};

interface Props extends Omit<TextFieldProps, 'type'> {}

export default NumberInput;
