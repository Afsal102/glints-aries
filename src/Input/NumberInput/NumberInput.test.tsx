import * as React from 'react';

import * as renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import NumberInput from './NumberInput';

const props = {
  label: 'Number',
  onChange: jest.fn().mockImplementation(event => event.target.value),
};

const inputValue = '12';

function setupNumberInput(otherProps?: any) {
  const { getByLabelText, ...utils } = render(
    <NumberInput
      label={props.label}
      onChange={props.onChange}
      {...otherProps}
    />
  );
  const numberInput = getByLabelText(props.label) as HTMLInputElement;
  return { numberInput, utils };
}

function setupNumberInputWithChangeEvent(otherProps?: any) {
  const { numberInput } = setupNumberInput(otherProps);
  userEvent.type(numberInput, inputValue);
  return { numberInput };
}

it('<NumberInput> should render with a label of Username', () => {
  const TextFieldSnapshot = renderer
    .create(<NumberInput label={props.label} />)
    .toJSON();
  expect(TextFieldSnapshot).toMatchSnapshot();
});

describe('when it is rendered', () => {
  it('should have an empty string as its value', () => {
    const { numberInput } = setupNumberInput();
    expect(numberInput.value).toEqual('');
  });
});

describe('when a value is entered', () => {
  it('should call onChange for each input', () => {
    setupNumberInputWithChangeEvent();
    expect(props.onChange).toHaveBeenCalledTimes(inputValue.length);
  });

  it('should return the input value when onChange is passed a function: event => event.target.value', () => {
    setupNumberInputWithChangeEvent();
    expect(props.onChange.mock.results.slice(-1)[0].value).toEqual(inputValue);
  });

  it('should display the correct value', () => {
    const { numberInput } = setupNumberInputWithChangeEvent();
    expect(numberInput.value).toEqual(inputValue);
  });
});
