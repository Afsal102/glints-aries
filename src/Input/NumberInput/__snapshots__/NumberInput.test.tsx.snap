// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NumberInput> should render correctly to match snapshot 1`] = `
<div
  className="TextFieldStyle__TextFieldContainer-sc-5sd02x-3 SzXyD aries-textfield"
>
  <input
    aria-label="Number"
    className="TextFieldStyle__TextFieldInput-sc-5sd02x-5 hytRJV"
    disabled={false}
    onKeyDown={[Function]}
    type="number"
  />
  <label
    className="TextFieldStyle__TextFieldLabel-sc-5sd02x-4 lmsuTp textfield-label"
    data-testid="textfield-label"
  >
    Number
  </label>
</div>
`;
